tests/unit_tests/test_internal_api.py

```diff
@@ -99,6 +99,218 @@ def test_internal_api_with_no_write_global_config_dir(tmp_path):
         os.chmod(tmp_path, 0o777)  # Allow the test runner to clean up.
 
 
+def test_fetch_orgs_and_org_entities_from_entity(mock_server):
+    """Test fetching organization entities from an entity."""
+    api = wandb.Api()
+
+    # Test team/org entity case
+    mock_server.gql.return_value = {
+        "entity": {
+            "organization": {
+                "name": "test-org",
+                "orgEntity": {"name": "test-org-entity"},
+            },
+            "user": None,
+        }
+    }
+    result = api._fetch_orgs_and_org_entities_from_entity("team-entity")
+    assert result == [("test-org-entity", "test-org")]
+
+    # Test personal entity with single org
+    mock_server.gql.return_value = {
+        "entity": {
+            "organization": None,
+            "user": {
+                "organizations": [
+                    {
+                        "name": "personal-org",
+                        "orgEntity": {"name": "personal-org-entity"},
+                    }
+                ]
+            },
+        }
+    }
+    result = api._fetch_orgs_and_org_entities_from_entity("personal-entity")
+    assert result == [("personal-org-entity", "personal-org")]
```

Comment:
suggestion: mentioned offline -- consider separating each of these cases into its own test function for better test isolation